src/tests/index.tests.js

```diff
@@ -0,0 +1,407 @@
+describe ('HW_4', function() {
+    describe ('ArrayList', function() {
+        describe('Initialization (init)', function() {
+            it('should be defined', function() {
+                assert.isDefined(ArrayList.prototype.init, 'Initialization process has been defined');
+            });
+
+            it('should be Warning (undefined)', function() {
+                const arr = undefined;
+                const expSize = 'Booom try again!';
+                const expArr = 'Booom try again!';
+
+                const actual = ArrayList.prototype.init(arr);
```

Comment:
this is not right testing flow, we agreed to test this type functions in indirect way